### PR TITLE
Nova/mimc

### PR DESCRIPTION
**Pull request by KiriosK**

### CODE REVIEW

File: nova/src/hash/circuit.rs [hunk 1]
```diff
@@ -0,0 +1,105 @@
+use crate::hash::Mimc;
+use r1cs::gadget::curve::PointAssignment;
+use r1cs::gadget::field::FieldAssignment;
+use r1cs::{CircuitDriver, R1cs};
+use zkstd::common::IntGroup;
+
+pub(crate) struct ZKMimc<const ROUND: usize, C: CircuitDriver> {
```
Comment (ashWhiteHat) [inline nova/src/hash/circuit.rs:7]: Considering name rule so far, MimcAssignment is fine

File: nova/Cargo.toml [hunk 2]
```diff
@@ -13,8 +13,6 @@ keywords = ["zkp", "blockchain", "elliptic-curve"]
 [dependencies]
 r1cs = { path = "../r1cs", default-features = false }
 zkstd = { path = "../zkstd", default-features = false }
-neptune = { version = "13.0.0", default-features = false }
-merlin = { version = "3.0", default-features = false }
 generic-array = { version = "0.14.7", default-features = false }
 bn-254 = { path = "../bn254", default-features = false }
 serde = { version = "1.0.102", default-features = false, features = ["derive"] }
```
Comment (ashWhiteHat) [inline nova/Cargo.toml:18]: I think we can remove ff, generic-array.
Do you check carefully?

File: nova/src/prover.rs [hunk 3]
```diff
@@ -30,16 +29,16 @@ impl<C: CircuitDriver> Prover<C> {
         r1cs: &R1cs<C>,
         relaxed_r1cs: &RelaxedR1cs<C>,
     ) -> (RelaxedR1csInstance<C>, RelaxedR1csWitness<C>, C::Affine) {
-        let mut transcript = Merlin::new(b"nova");
+        let mut transcript = MimcRO::<322, C::Base>::default();
```
Comment (ashWhiteHat) [inline nova/src/prover.rs:32]: Where does this 322 come from?

File: nova/src/verifier.rs [hunk 4]
```diff
@@ -15,12 +14,12 @@ impl<C: CircuitDriver> Verifier<C> {
         r1cs: &R1cs<C>,
         relaxed_r1cs: &RelaxedR1cs<C>,
     ) -> RelaxedR1csInstance<C> {
-        let mut transcript = Merlin::new(b"nova");
+        let mut transcript = MimcRO::<322, C::Base>::default();
```
Comment (ashWhiteHat) [inline nova/src/verifier.rs:17]: ditto

**Review by ashWhiteHat — CHANGES_REQUESTED** [review]

File: nova/src/prover.rs (hunk 3 above)
Comment (KiriosK) [reply nova/src/prover.rs:32]: The 5.1 part describes how the number of rounds should be calculated. If we have an `n` bit field.
We use the Feistel mode with k equals to zero, so it's doubled.
https://eprint.iacr.org/2016/492.pdf#page=17

File: nova/src/hash/circuit.rs (hunk 1 above)
Comment (KiriosK) [reply nova/src/hash/circuit.rs:7]: Renamed

File: nova/src/prover.rs (hunk 3 above)
Comment (ashWhiteHat) [reply nova/src/prover.rs:32]: Can you replace magic number 322 with constant variable with understandable naming?

Comment (KiriosK) [reply nova/src/prover.rs]: Done

**Pull request merged**